openmdao/solvers/nonlinear/nonlinear_block_gs.py

```diff
@@ -26,18 +26,95 @@ def _setup_solvers(self, system, depth):
         if len(system._subsystems_allprocs) != len(system._subsystems_myproc):
             raise RuntimeError('Nonlinear Gauss-Seidel cannot be used on a parallel group.')
 
+    def _declare_options(self):
+        """
+        Declare options before kwargs are processed in the init method.
+        """
+        
+        self.options.declare('use_aitken', type_=bool, default=False,
+                             desc='set to True to use Aitken relaxation')
+        self.options.declare('aitken_min_factor', default=0.1,
+                             desc='lower limit for Aitken relaxation factor')
+        self.options.declare('aitken_max_factor', default=1.5,
+                             desc='upper limit for Aitken relaxation factor')
+
+    def _iter_initialize(self):
+        """
+        Perform any necessary pre-processing operations.
+
+        Returns
+        -------
+        float
+            initial error.
+        float
+            error at the first iteration.
+        """
+        self._aitken_work1 = self._system._outputs._clone()
```

Comment:
All of the new code here and in `_iter_execute` should be under an `if self.options['use_aitken'] is True` block. Don't want to be cloning the output vector and doing all the other calculation if we aren't using aitken.

Reply:
Thanks, I have added the IF statements.